scripts/04-find-patterns.py

```
@@ -64,33 +64,33 @@ def _calculate_patterns_and_metrics(
             for pattern_info in patterns_info:
                 try:
-                    pattern = pattern_info["make"]()
+                    pattern = pattern_info['make']()
                     pattern_result = pattern.value(component_ast)
-                    calculation_result[pattern_info["code"]] = len(pattern_result)
-                    calculation_result["lines_" + pattern_info["code"]] = pattern_result
+                    calculation_result[pattern_info['code']] = len(pattern_result)
+                    calculation_result['lines_' + pattern_info['code']] = pattern_result
                 except Exception as cause:
-                    raise FileProcessingError(file_path, pattern_info["name"], cause)
+                    raise FileProcessingError(file_path, pattern_info['name'], cause)
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)